src/app/qgsmaptooladdfeature.cpp

```diff
@@ -175,6 +175,14 @@ void QgsMapToolAddFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
         g = QgsGeometry::fromPoint( savePoint );
       }
 
+      // Use default Z-value if coordinates is enabled.
+      if (QgsWkbTypes::hasZ(layerWKBType))
+          g.geometry()->addZValue(defaultZValue());
+
+      // Use default M-value if coordinates is enabled.
+      if (QgsWkbTypes::hasM(layerWKBType))
+          g.geometry()->addMValue(0.0 /* No default M-value yet. */);
```

Comment:
wouldn't be better to define `defaultMValue()` ?

Reply:
Better use `std::numeric_limits<double>::quiet_NaN()`